pinot-common/src/main/java/org/apache/pinot/common/utils/UploadedRealtimeSegmentName.java

```diff
@@ -0,0 +1,199 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.utils;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+import java.util.Arrays;
+import java.util.Objects;
+import javax.annotation.Nullable;
+import org.apache.commons.lang3.StringUtils;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+
+
+/**
+ * Class to represent segment names like: uploaded__{tableName}__{partitionId}__{sequenceId}__{creationTime}__{
+ * optionalSuffix}
+ *
+ * <p>This naming convention is adopted to represent a segment uploaded to a realtime table. The naming
+ * convention has been kept similar to {@LLCSegmentName} to but differentiates between stream generated LLCSegments
+ * based on the prefix "uploaded" and an optional suffix.
+ */
+public class UploadedRealtimeSegmentName implements Comparable<UploadedRealtimeSegmentName> {
```

Comment:
It might be helpful to that "Not like the seqId in LLCSemgentName, the seqId here is not ensured to monotonically increase but more for avoiding name conflicts when uploading segments with same ctime suffix. As to why using same ctime, it would allow one to overwrite previously segments more easily."

Reply:
Agreed, the sequence numbers are externally generated and may not have same meaning.

Reply:
typo: s/uplaoded/uploaded